app/models.py

```diff
@@ -930,36 +931,53 @@ def applications_closed_at(self):
         if self.published_at is None:
             return None
 
-        # Set time to 23:59:59 same day and add full number of days before application closes
         published_day = self.published_at.replace(hour=23, minute=59, second=59, microsecond=0)
-        closing_time = published_day + timedelta(days=self.APPLICATIONS_OPEN_DAYS)
+        requirements_length = self.data.get('requirementsLength')
+        data = {
+            'publishedAt': published_day,
+            'requirementsLength': requirements_length
+        }
+        closing_date = get_publishing_dates(data)['closing_date']
 
-        return closing_time
+        return closing_date
 
     @applications_closed_at.expression
     def applications_closed_at(cls):
-        return func.date_trunc('day', cls.published_at) + sql_cast(
-            '%d days 23:59:59' % cls.APPLICATIONS_OPEN_DAYS, INTERVAL
-        )
+        return sql_case([
+            (cls.data['requirementsLength'].astext == '1 week', func.date_trunc('day', cls.published_at) + sql_cast(
+                '7 days 23:59:59', INTERVAL)),
+            ], else_=func.date_trunc('day', cls.published_at) + sql_cast(
+                '14 days 23:59:59', INTERVAL))
 
     @hybrid_property
     def clarification_questions_closed_at(self):
         if self.published_at is None:
             return None
 
-        # Set time to 23:59:59 same day and add full number of days before questions close
         published_day = self.published_at.replace(hour=23, minute=59, second=59, microsecond=0)
-        closing_time = published_day + timedelta(days=self.CLARIFICATION_QUESTIONS_OPEN_DAYS)
+        requirements_length = self.data.get('requirementsLength')
+        data = {
+            'publishedAt': published_day,
+            'requirementsLength': requirements_length
+        }
+        closing_date = get_publishing_dates(data)['questions_close']
 
-        return closing_time
+        return closing_date
 
     @hybrid_property
     def clarification_questions_published_by(self):
         if self.published_at is None:
             return None
 
-        # All clarification questions should be published N days before brief closes
-        return self.applications_closed_at - timedelta(days=self.CLARIFICATION_QUESTIONS_PUBLISHED_DAYS)
+        published_day = self.published_at.replace(hour=23, minute=59, second=59, microsecond=0)
+        requirements_length = self.data.get('requirementsLength')
+        data = {
+            'publishedAt': published_day,
+            'requirementsLength': requirements_length
+        }
+        closing_date = get_publishing_dates(data)['answers_close']
+
+        return closing_date
```

Comment:
@pcraig3 have extracted the code out to a separate function in the model. What do you think?